### PR TITLE
returns HTTP 201 on resource creation

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**:
it turned out that Write method writes the data to the connection.
If WriteHeader method has not yet been called, Write calls
WriteHeader(http.StatusOK) before writing the data.

setting desired HTTP header before calling json.Encode resolves the issue.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/handler.go [hunk 1]
```diff
@@ -67,14 +67,10 @@ func StatusOK(res http.ResponseWriter, _ *http.Request) {
...
-func createStatusResource(f func(context.Context, http.ResponseWriter, interface{}) error) func(context.Context, http.ResponseWriter, interface{}) error {
+func setStatusCreatedHeader(f func(context.Context, http.ResponseWriter, interface{}) error) func(context.Context, http.ResponseWriter, interface{}) error {
 	return func(ctx context.Context, r http.ResponseWriter, i interface{}) error {
-		err := f(ctx, r, i)
-		if err != nil {
-			return err
-		}
 		r.WriteHeader(http.StatusCreated)
-		return nil
+		return f(ctx, r, i)
```
Comment (mrIncompetent) [inline api/pkg/handler/handler.go:73]: changing the order that way does still return a 500 when having an error?

Comment (p0lyn0mial) [reply api/pkg/handler/handler.go:73]: let me verify that.

Comment (p0lyn0mial) [reply api/pkg/handler/handler.go:73]: it works, `go-kit` special cases failures and will actually call a different encoder.

```
response, err := s.e(ctx, request)
	if err != nil {
		s.logger.Log("err", err)
		s.errorEncoder(ctx, err, w)
		return
	}
```

`s.e(ctx,request)` being the endpoint

source code:
https://github.com/go-kit/kit/blob/master/transport/http/server.go#L110

**Comment by mrIncompetent** [comment]
Thanks for verifying!

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**